### ISSUE TITLE
[BUG] No kernel boot log when SOF FW loading fails

### ISSUE THREAD

**Issue by fredoh9** [thesofproject/sof-test#1036]
**Describe the bug**
No kernel log is captured in test result. This is side effect after https://github.com/thesofproject/sof-test/pull/1033 merged.
'start_test' will be called before the every test case, if it fails there, too early to capture the log

**To Reproduce**
This is one of example, FW loading is failed
https://sof-ci.01.org/sofpr/PR7584/build7585/devicetest/index.html?model=ADLP_RVP_SDW_IPC4ZPH&testcase=verify-kernel-boot-log

**Expected behavior**
kernel log should be available on the failure

**Comment by marc-hb** [comment]
> I think @lyakh found an issue with this new feature: kernel logs seem to be missing when the firmware does not boot. Example: https://sof-ci.01.org/sofpr/PR7584/build7585/devicetest/index.html

In jenkins logs generic_test/13429/
```
10:58:40  cmd: scp -r ubuntu@sh-adlp-rvp-sdw-01.sh.intel.com:~/sof-test/test-case/../logs/verify-firmware-presence/last/* results/2023-05-12-03:55:59/sh-adlp-rvp-sdw-01/verify-firmware-presence
10:58:40  return code: 1
10:58:40  cmd output: scp: /home/ubuntu/sof-test/test-case/../logs/verify-firmware-presence/last/*: No such file or directory
```

This is very likely because the new code runs "too early".

_Originally posted by @marc-hb in https://github.com/thesofproject/sof-test/issues/1033#issuecomment-1548052202_

**Comment by marc-hb** [comment]
Fix submitted: #1191

**Comment by marc-hb** [comment]
Should be good, will re-open if any issue.
